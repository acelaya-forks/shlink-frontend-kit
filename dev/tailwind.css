@import 'tailwindcss' prefix(tw) important;
@import '../src/tailwind.preset.css';
