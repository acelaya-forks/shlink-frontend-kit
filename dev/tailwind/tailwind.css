@import 'tailwindcss' prefix(tw) important;
@import '../../src/tailwind/tailwind.preset.css';
