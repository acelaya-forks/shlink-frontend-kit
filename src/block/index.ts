export * from './Message';
export * from './Result';
export * from './SimpleCard';
