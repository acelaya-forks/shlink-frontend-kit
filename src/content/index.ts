export * from './Details';
export * from './Listbox';
export * from './Table';
