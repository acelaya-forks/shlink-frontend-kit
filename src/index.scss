@import './utils/ResponsiveTable';
@import './theme/theme';
