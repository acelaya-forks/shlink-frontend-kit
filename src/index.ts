export * from './block';
export * from './form';
export * from './hooks';
export * from './navigation';
export * from './ordering';
