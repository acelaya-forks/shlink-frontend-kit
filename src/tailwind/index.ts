export * from './content';
export * from './feedback';
export * from './form';
export * from './helpers';
export * from './navigation';
export * from './surfaces';
export * from './utils';
export * from './types';
