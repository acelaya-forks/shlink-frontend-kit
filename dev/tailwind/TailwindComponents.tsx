import type { FC } from 'react';
import { Route, Routes } from 'react-router';
import { DetailsPage } from './content/DetailsPage';
import { TablePage } from './content/TablePage';
import { MessagePage } from './feedback/MessagePage';
import { ModalDialogPage } from './feedback/ModalDialogPage';
import { ResultPage } from './feedback/ResultPage';
import { TooltipPage } from './feedback/TooltipPage';
import { ButtonsPage } from './form/ButtonsPage';
import { InputsPage } from './form/InputsPage';
import { SearchComboboxPage } from './form/SearchComboboxPage';
import { SearchInputPage } from './form/SearchInputPage';
import { NavPillsPage } from './navigation/NavPillsPage';
import { PaginatorPage } from './navigation/PaginatorPage';
import { CardsPage } from './surfaces/CardsPage';
import { UtilsPage } from './utils/UtilsPage';

export const TailwindComponents: FC = () => {
  return (
    <Routes>
      <Route path="/" element={<h2 className="text-center">Select component - Tailwind</h2>} />
      <Route path="/form/inputs" element={<InputsPage />} />
      <Route path="/form/search-input" element={<SearchInputPage />} />
      <Route path="/form/search-combobox" element={<SearchComboboxPage />} />
      <Route path="/form/buttons" element={<ButtonsPage />} />
      <Route path="/surfaces/cards" element={<CardsPage />} />
      <Route path="/content/tables" element={<TablePage />} />
      <Route path="/content/details" element={<DetailsPage />} />
      <Route path="/navigation/paginator" element={<PaginatorPage />} />
      <Route path="/navigation/nav-pills">
        <Route path="" element={<NavPillsPage />} />
        <Route path="*" element={<NavPillsPage />} />
      </Route>
      <Route path="/feedback/dialogs" element={<ModalDialogPage />} />
      <Route path="/feedback/result" element={<ResultPage />} />
      <Route path="/feedback/message" element={<MessagePage />} />
      <Route path="/feedback/tooltip" element={<TooltipPage />} />
      <Route path="/utils" element={<UtilsPage />} />
      <Route path="*" element={<h2 className="text-center">Not found - Tailwind</h2>} />
    </Routes>
  );
};
