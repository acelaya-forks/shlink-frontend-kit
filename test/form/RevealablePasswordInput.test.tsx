import { screen } from '@testing-library/react';
import type { RevealablePasswordInputProps } from '../../src/tailwind';
import { RevealablePasswordInput } from '../../src/tailwind';
import { checkAccessibility } from '../__helpers__/accessibility';
import { renderWithEvents } from '../__helpers__/setUpTest';

describe('<RevealablePasswordInput />', () => {
  const setUp = (props: RevealablePasswordInputProps = {}) => renderWithEvents(
    <>
      <RevealablePasswordInput {...props} />
      <button data-testid="alt-button">Some other focusable element</button>
    </>,
  );

  it('passes a11y checks', () => checkAccessibility(setUp({ 'aria-label': 'The password' })));

  it('toggles password when clicking reveal button', async () => {
    const { user } = setUp();

    expect(screen.getByTestId('input')).toHaveAttribute('type', 'password');
    await user.click(screen.getByLabelText('Show password'));
    expect(screen.getByTestId('input')).toHaveAttribute('type', 'text');
    await user.click(screen.getByLabelText('Hide password'));
    expect(screen.getByTestId('input')).toHaveAttribute('type', 'password');
  });

  it('hides password when focus moves away', async () => {
    const { user } = setUp();

    expect(screen.getByTestId('input')).toHaveAttribute('type', 'password');
    await user.click(screen.getByLabelText('Show password'));
    expect(screen.getByTestId('input')).toHaveAttribute('type', 'text');
    await user.click(screen.getByTestId('alt-button'));
    expect(screen.getByTestId('input')).toHaveAttribute('type', 'password');
  });

  it.each([
    'sm' as const,
    'md' as const,
    'lg' as const,
  ])('renders different classes depending on the size', (size) => {
    setUp({ size });
    expect(screen.getByTestId('input')).toMatchSnapshot();
  });
});
