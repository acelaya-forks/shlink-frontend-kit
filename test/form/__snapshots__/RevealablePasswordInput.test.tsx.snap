// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<RevealablePasswordInput /> > renders different classes depending on the size 1`] = `
<input
  class="tw:outline-none tw:w-full tw:focus-ring tw:px-2 tw:py-1 tw:text-sm tw:rounded-md tw:border tw:border-lm-input-border tw:dark:border-dm-input-border tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input tw:pr-8"
  data-testid="input"
  type="password"
/>
`;

exports[`<RevealablePasswordInput /> > renders different classes depending on the size 2`] = `
<input
  class="tw:outline-none tw:w-full tw:focus-ring tw:px-3 tw:py-1.5 tw:rounded-md tw:border tw:border-lm-input-border tw:dark:border-dm-input-border tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input tw:pr-10"
  data-testid="input"
  type="password"
/>
`;

exports[`<RevealablePasswordInput /> > renders different classes depending on the size 3`] = `
<input
  class="tw:outline-none tw:w-full tw:focus-ring tw:px-4 tw:py-2 tw:text-xl tw:rounded-md tw:border tw:border-lm-input-border tw:dark:border-dm-input-border tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input tw:pr-10"
  data-testid="input"
  type="password"
/>
`;
